diff --git a/cmake/find-modules/FindOIS.cmake b/cmake/find-modules/FindOIS.cmake
index 55fa580ae..55b4d32d1 100644
--- a/cmake/find-modules/FindOIS.cmake
+++ b/cmake/find-modules/FindOIS.cmake
@@ -35,7 +35,7 @@
 # * OIS::OIS
 #
 
-find_path(OIS_INCLUDE_DIR OIS.h PATH_SUFFIXES OIS)
+find_path(OIS_INCLUDE_DIR OIS.h PATH_SUFFIXES ois)
 find_library(OIS_LIBRARY OIS)
 
 set(OIS_INCLUDE_DIRS ${OIS_INCLUDE_DIR})
diff --git a/source/main/gui/DashBoardManager.cpp b/source/main/gui/DashBoardManager.cpp
index 06f0f528c..9f02487ad 100644
--- a/source/main/gui/DashBoardManager.cpp
+++ b/source/main/gui/DashBoardManager.cpp
@@ -867,7 +867,7 @@ bool DashBoard::parseLink(std::string& linkArgs, int& linkID, char& condition, f
 void DashBoard::loadLayoutRecursive(MyGUI::WidgetPtr w)
 {
     std::string name = w->getName();
-    std::string debug = w->getUserString("debug");
+    std::string debug = std::string(w->getUserString("debug"));
 
     // make it unclickable
     w->setUserString("interactive", "0");
@@ -927,7 +927,7 @@ void DashBoard::loadLayoutRecursive(MyGUI::WidgetPtr w)
     std::string formatNumStr = "format";
     std::string directionNumStr = "direction";
 
-    std::string anim = w->getUserString(animNumStr);
+    std::string anim = std::string(w->getUserString(animNumStr));
     std::string linkArgs;
     if (anim != "")
     {
@@ -1040,11 +1040,11 @@ void DashBoard::loadLayoutRecursive(MyGUI::WidgetPtr w)
                     ctrl.graphicalAnimation.condition = condition;
                     ctrl.graphicalAnimation.conditionArgument = conditionArgument;
 
-                    String texture = w->getUserString(textureNumStr);
+                    String texture = std::string(w->getUserString(textureNumStr));
                     if (!texture.empty())
                         strncpy(ctrl.graphicalAnimation.texture, texture.c_str(), sizeof ctrl.graphicalAnimation.texture);
 
-                    String format = w->getUserString(formatNumStr);
+                    String format = std::string(w->getUserString(formatNumStr));
                     if (!format.empty())
                         strncpy(ctrl.graphicalAnimation.format, format.c_str(), sizeof ctrl.graphicalAnimation.format);
 
@@ -1143,12 +1143,12 @@ void DashBoard::loadLayoutRecursive(MyGUI::WidgetPtr w)
                         ctrl.geometricAnimationCount++;
 
                         // parse more attributes
-                        geometricAnim.wmin = StringConverter::parseReal(w->getUserString(minNumStr));
-                        geometricAnim.wmax = StringConverter::parseReal(w->getUserString(maxNumStr));
-                        geometricAnim.vmin = StringConverter::parseReal(w->getUserString(vminNumStr));
-                        geometricAnim.vmax = StringConverter::parseReal(w->getUserString(vmaxNumStr));
+                        geometricAnim.wmin = StringConverter::parseReal(std::string(w->getUserString(minNumStr)));
+                        geometricAnim.wmax = StringConverter::parseReal(std::string(w->getUserString(maxNumStr)));
+                        geometricAnim.vmin = StringConverter::parseReal(std::string(w->getUserString(vminNumStr)));
+                        geometricAnim.vmax = StringConverter::parseReal(std::string(w->getUserString(vmaxNumStr)));
 
-                        String direction = w->getUserString(directionNumStr);
+                        String direction = std::string(w->getUserString(directionNumStr));
                         if (direction == "right")
                             geometricAnim.direction = DIRECTION_RIGHT;
                         else if (direction == "left")
diff --git a/source/main/gui/RTTLayer.cpp b/source/main/gui/RTTLayer.cpp
index 6942ca30e..60286173e 100644
--- a/source/main/gui/RTTLayer.cpp
+++ b/source/main/gui/RTTLayer.cpp
@@ -32,8 +32,8 @@ void RTTLayer::deserialization(xml::ElementPtr _node, Version _version)
     MyGUI::xml::ElementEnumerator propert = _node->getElementEnumerator();
     while (propert.next("Property"))
     {
-        const std::string& key = propert->findAttribute("key");
-        const std::string& value = propert->findAttribute("value");
+        const std::string& key = std::string(propert->findAttribute("key"));
+        const std::string& value = std::string(propert->findAttribute("value"));
         if (key == "TextureSize")
             setTextureSize(utility::parseValue<IntSize>(value));
         if (key == "TextureName")
